doc/modules/mod_stream_management.md

```diff
@@ -1,21 +1,100 @@
 ### Module Description
-Enables [XEP-0198: Stream Management](http://xmpp.org/extensions/xep-0198.html). 
-It is implemented mostly in `ejabberd_c2s`. 
-This module is just a "starter", to supply the configuration values to new client connections.
-It also provides a basic session table API and adds a new stream feature.
+
+Enables [XEP-0198: Stream Management](http://xmpp.org/extensions/xep-0198.html).
+Most of the logic regarding session resumption and acknowledgement is implemented in `ejabberd_c2s`,
+while the management of the session tables and configuration is implemented in
+`mod_stream_management`.
+
+### In `ejabberd_c2s`
+
+The record `#state{}` in the `ejabberd_c2s` `gen_fsm` server keeps fields like:
+
+```erlang
+stream_mgmt = false, %% whether SM is enabled, used in pattern matching inside `ejabberd_c2s`
+stream_mgmt_in = 0, %% amount of msgs on the server and not acked by the user (server's <h>)
+stream_mgmt_id, %% the mod_stream_management:smid() unique identifier
+stream_mgmt_out_acked = 0, %% messages delivered to the user, and acked by the user (user's <h>)
+stream_mgmt_buffer = [], %% buffered stanzas not yet acked by the user
+stream_mgmt_buffer_size = 0, %% amount of messages buffered for the user
+stream_mgmt_buffer_max = ?STREAM_MGMT_CACHE_MAX, %% server's capacity for buffering
+stream_mgmt_ack_freq = ?STREAM_MGMT_ACK_FREQ, %% how often the server requests acks
+stream_mgmt_resume_timeout = ?STREAM_MGMT_RESUME_TIMEOUT, %% resumption timeout
+stream_mgmt_resume_tref, %% a ref() to pattern-match a given timeout
+stream_mgmt_resumed_from, %% a ejabberd_sm:sid() to keep identifiying the old session
+stream_mgmt_constraint_check_tref, %% another ref() for a timeout, this time for buffer_full check
+```
+
+### In `mod_stream_management`
+
+This module is just a "starter", to supply the configuration values to new client connections. It
+also provides a basic session table API and adds a new stream feature.
+
+At a bare minimum, this module keeps the config values in its `gen_mod` records, and it keeps a
+mnesia table defined as follows:
+
+```erlang
+-record(sm_session,
+        {smid :: smid(),
+         sid :: ejabberd_sm:sid()
+        }).
+```
+
+Where `smid` is a unique identifier — in this case a random binary, and `sid` is an opaque session
+identifier from `ejabberd_sm`, which is needed to find the previous session we want to resume from.
+This module implements hooks that run on connection removals and session cleanups, in order to clean
+records from a dying session; and it also implements registration callbacks, used in `ejabberd_c2s`
+when a session is registered for resumption.
+
+In order to be compliant with the XEP version 1.6, where a server can do the best effort to give the
```

Comment:
XEP version 1.6 requires the server to attempt giving the user the value of the server's `<h>` when a session timed out and cannot be resumed anymore.
To be compliant with it, there's a second optional table:

alt:

XEP version 1.6 requires the server to give the user the value of the server's `<h>` when a session timed out and cannot be resumed anymore.
To be compliant with it, there's a second optional table: